integration-tests/build.rs

```diff
@@ -0,0 +1,247 @@
+use ethers::{
+    abi::Contract,
+    core::types::Bytes,
+    solc::{
+        artifacts::{CompactContractRef, Error as SolcArtifactError},
+        CompilerInput, CompilerOutput, EvmVersion, Solc,
+    },
+};
+use ethers_contract_abigen::Abigen;
+use serde::{Deserialize, Serialize};
+use std::{fs::File, path::Path};
+use thiserror::Error;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CompiledContract {
+    /// Contract path
+    pub path: String,
+    /// Contract name
+    pub name: String,
+    /// ABI
+    pub abi: Contract,
+    /// Bytecode
+    pub bin: Bytes,
+    /// Runtime Bytecode
+    pub bin_runtime: Bytes,
+}
+
+#[derive(Error, Debug)]
+enum BuildError {
```

Comment:
I think you did some really nice work with this error type here!  Nevertheless I see that there's no error recovery or error handling in here: whenever an error is found, the build process stops with the error message.  For this reason I think it's also OK to just use `expect()` with a nice message of the failure, so that a panic happens on an error.
This is just a personal thought to reduce the amount of work in future code!  I'm not saying to change this (I think using the error type here is really clear and nice), but maybe you can skip some work just using `expect()` or even `panic()` in the context of `build.rs`.

Reply:
Sorry, it was my bad to suggest changing to this style. The PR originally handled errors with `expect`s and `panic`s.
I agree with you that `build.rs` is a good place just to use panics, and I also agree that the error type looks clearer, so no change is required.

Reply:
I think we can let it be for now. In any case CC's suggestion was a very useful direction for me on using the error type for cases were we want to actually handle the error itself!